src/main/java/aztech/modern_industrialization/machines/models/MachineBakedModel.java

```diff
@@ -70,12 +71,18 @@ public boolean isVanillaAdapter() {
     @Override
     public void emitBlockQuads(BlockRenderView blockRenderView, BlockState blockState, BlockPos blockPos, Supplier<Random> supplier,
             RenderContext renderContext) {
-        if (blockRenderView instanceof RenderAttachedBlockView) {
-            RenderAttachedBlockView bv = (RenderAttachedBlockView) blockRenderView;
+        if (blockRenderView instanceof RenderAttachedBlockView bv) {
             Object attachment = bv.getBlockEntityRenderAttachment(blockPos);
-            if (attachment instanceof MachineModelClientData) {
-                MachineModelClientData clientData = (MachineModelClientData) attachment;
-                MachineCasingModel casing = clientData.casing == null ? defaultCasing : clientData.casing.mcm;
+            if (attachment instanceof MachineModelClientData clientData) {
+                MachineCasingModel casing;
+                if (clientData.casing == null) {
+                    casing = defaultCasing;
+                    System.out.println("defualt");
+                } else {
+                    casing = clientData.casing.mcm;
+                    System.out.println("client");
+                    System.out.println(clientData.casing.mcm.getMesh());
+                }
```

Comment:
Debug prints. :smile:

Reply:
oh god
i was about to take them out but then got a dm